### ISSUE TITLE
Large filenumber prevents program from running

### ISSUE THREAD

**Issue by djevo1** [worldveil/dejavu#112]
I setup a dejavu python script to run across and index a directory of mp4s. There are thousands of files within this directory and it appears to get stuck. Is there a way around this issue. This was left running for 12 hours with no progress made.

**Comment by atlantis0** [comment]
I have run into similar issues a few days ago. Issue might be memory. I can't confirm honestly. The mp4s, how big are they in MB (individually)?

**Comment by djevo1** [comment]
Around 100mb each or slightly larger

---Sent from Boxer | http://getboxer.com

On May 9, 2017 at 5:03:49 AM PDT, Ayele <notifications@github.com> wrote:I have run into similar issues a few days ago. Issue might be memory. I can't confirm honestly. The mp4s, how big are they in MB (individually)?  —You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.           {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/worldveil/dejavu","title":"worldveil/dejavu","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/worldveil/dejavu"}},"updates":{"snippets":[{"icon":"PERSON","message":"@atlantis0 in #112: I have run into similar issues a few days ago. Issue might be memory. I can't confirm honestly. The mp4s, how big are they in MB (individually)?"}],"action":{"name":"View Issue","url":"https://github.com/worldveil/dejavu/issues/112#issuecomment-300142629"}}}

**Comment by atlantis0** [comment]
Yes. Confirmed my suspicions. Try to load your mp4 using the following lib http://pydub.com/. dejavu uses it. You will run into the same issue. I mitigated the error by splitting large media files

**Comment by djevo1** [comment]
In this specific case we cant split files. I wrote a wrapper that uses walk to loop over the directory and use the dejavu index single file to index. The only issue I am having now is a mysql lock timeout issue.

---Sent from Boxer | http://getboxer.com

On May 9, 2017 at 9:41:52 AM PDT, Ayele <notifications@github.com> wrote:Yes. Confirmed my suspicions. Try to load your mp4 using the following lib http://pydub.com/. dejavu uses it. You will run into the same issue. I mitigated the error by splitting large media files  —You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.           {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/worldveil/dejavu","title":"worldveil/dejavu","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/worldveil/dejavu"}},"updates":{"snippets":[{"icon":"PERSON","message":"@atlantis0 in #112: Yes. Confirmed my suspicions. Try to load your mp4 using the following lib http://pydub.com/. dejavu uses it. You will run into the same issue. I mitigated the error by splitting large media files"}],"action":{"name":"View Issue","url":"https://github.com/worldveil/dejavu/issues/112#issuecomment-300225910"}}}

**Comment by djevo1** [comment]
It appears the lock is being caused by dejavu, below is the code.

```
import sys
import os
#fileloc = sys.argv[-1]
from dejavu import Dejavu
config = {
        "database": {
                "host": "10.1.10.202",
                "user": "e",
                "passwd": "passwd",
                "db": "finger",
                }
        }

def indexdata (varg1, varg2):
    djv = Dejavu(config)
    f = open('/Users/svMarx/Documents/music-index.txt', 'a')
    section = varg1.split('_')
    if section[2] == 'XZ':
        djv.fingerprint_file(os.path.join("/Volumes/ETC_2015/BU-MP4/BU-MP4-VJ-Pro-Smash", varg1))
        f.write(varg2 + "\n")
        return;
    elif section[2] == 'SE':
        djv.fingerprint_file(os.path.join("/Volumes/ETC_2015/BU-MP4/BU-MP4-VJ-Pro-Smash", varg1))
        f.write(varg2 + "\n")
        return;
    elif section[2] == 'IO':
        djv.fingerprint_file(os.path.join("/Volumes/ETC_2015/BU-MP4/BU-MP4-VJ-Pro-Smash", varg1))
        f.write(varg2 + "\n")
        return;
    elif section[2] == 'CB':
        djv.fingerprint_file(os.path.join("/Volumes/ETC_2015/BU-MP4/BU-MP4-VJ-Pro-Smash", varg1))
        f.write(varg2 + "\n")
        return;

for file in os.listdir("/Volumes/ETC_2015/BU-MP4/BU-MP4-VJ-Pro-Smash"):
    if file.endswith(".mp4"):
        section = file.split('_')
        music = section[1]
        datafile =  open('/Users/svMarx/Documents/music-index.txt', 'r')
        lines = datafile.read()
        name = lines.find(music)
        if name == -1:
            print music
            indexdata(varg1 = file,varg2 = music)
        else:
            continue

#print djv.db.get_num_fingerprints()
#quit()
```

**Comment by branford19** [comment]
Use a larger swapfile.
